### PR TITLE
Fix auditbeat test runs

### PR DESCRIPTION
**Pull request by mindbat**
Digging into a series of test failures on CI uncovered the fact that auditbeat needs AUDIT_CONTROL: https://github.com/elastic/beats/blob/d3769b5fae7f91ef9d064fa2a867ad6e2e8908f2/auditbeat/module/auditd/audit_linux.go#L189

(and host PID mode, for docker)

This PR adds the needed options to the docker-compose template, and the `docker run` used by the tests. Also tweaks one test that was checking for a pid of 1, which won't be true for auditbeat anymore.

The question of how the tests for auditbeat passed previously is left as an exercise for the reader.

### CODE REVIEW

File: tests/test_process.py [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 
 def test_process_is_pid_1(beat):
-    assert beat.process.pid == 1
+    if beat.name == 'auditbeat':
+        assert beat.process.pid >= 1
```
Comment (jarpy) [inline tests/test_process.py]: Is there ever a case where a process could have a PID that is not `>= 1`?

Did you mean `> 1`?

File: templates/docker-compose.yml.j2 [hunk 2]
```diff
@@ -12,6 +12,12 @@ services:
       - /var/log:/mnt/log:ro
     {%- endif %}
 
+    {%- if beat == 'auditbeat' %}
+    cap_add:
+      - AUDIT_CONTROL
+    pid: "host"
```
Comment (jarpy) [inline templates/docker-compose.yml.j2]: For better or worse, we don't explicitly quote the other strings in this YAML, so let's not here.

Comment (mindbat) [reply templates/docker-compose.yml.j2]: gotcha. fixed in bd91cca

File: tests/test_process.py (hunk 1 above)
Comment (mindbat) [reply tests/test_process.py]: excellent point. fixed in 73b936e

**Comment by jarpy** [comment]
It is difficult to express the full extent of my gratitude for this patch, which you so elegantly extracted from the inky depths of the rabbit-hole.

**Pull request merged**